Tangem/App/Services/WalletConnect/WalletConnectService/WalletConnectV1Service.swift

```diff
@@ -278,12 +220,62 @@ extension WalletConnectService: WalletConnectSessionController {
         }
     }
 
+}
+
+// MARK: - WalletConnectURLHandler
+extension WalletConnectV1Service: WalletConnectURLHandler {
     func canHandle(url: String) -> Bool {
         WCURL(url) != nil
     }
+
+    @discardableResult
+    func handle(url: URL) -> Bool {
+        guard let extracted = extractWcUrl(from: url) else { return false }
+
+        guard let wcUrl = WCURL(extracted.url) else { return false }
+
+        DispatchQueue.global().asyncAfter(deadline: .now() + extracted.handleDelay) {
+            self.connect(to: wcUrl)
+        }
+
+        return true
+    }
+
+    @discardableResult
+    func handle(url: String) -> Bool {
+        guard let url = URL(string: url) else { return false }
+
+        return handle(url: url)
+    }
+
+    private func extractWcUrl(from url: URL) -> ExtractedWCUrl? {
+        let absoluteStr = url.absoluteString
+        if canHandle(url: absoluteStr) {
+            return (url: absoluteStr, handleDelay: 0)
+        }
+
+        let uriPrefix = "uri="
+        let wcPrefix = "wc:"
```

Comment:
в constants может в шапку класса

```suggestion
private enum Constants {
        static let uriPrefix = "uri="
        static let wcPrefix = "wc:"
        }
```

тип такого)

Reply:
не хотел выносить то что нигде больше не используется. В общем-то без разницы.

Reply:
готово, сделал `ServiceConstants` потому что константы на уровне приложения уже есть